packages/js/tests/testHelpers/createDomTestData.js

```diff
@@ -0,0 +1,254 @@
+/* eslint-disable max-statements */
+
+/**
+ * Creates category elements.
+ *
+ * @returns {{category1: HTMLInputElement, category2: HTMLInputElement, category3: HTMLInputElement,
+ * category4: HTMLInputElement, category5: HTMLInputElement, allCategories: HTMLDivElement,
+ * mostUsedCategories: HTMLDivElement}} The elements created for categories.
+ */
+const createCategoryElements = () => {
+	// Create regular category elements.
+	const cat1 = document.createElement( "input" );
+	cat1.setAttribute( "type", "checkbox" );
+	cat1.setAttribute( "value", "1" );
+	const cat1Name = document.createTextNode( "cat1" );
+
+	const parentCat1 = document.createElement( "label" );
+	parentCat1.appendChild( cat1 );
+	parentCat1.appendChild( cat1Name );
+
+	const cat2 = document.createElement( "input" );
+	cat2.setAttribute( "type", "checkbox" );
+	cat2.setAttribute( "value", "2" );
+	const cat2Name = document.createTextNode( "cat2" );
+
+	const parentCat2 = document.createElement( "label" );
+	parentCat2.appendChild( cat2 );
+	parentCat2.appendChild( cat2Name );
+
+	const cat3 = document.createElement( "input" );
+	cat3.setAttribute( "type", "checkbox" );
+	cat3.setAttribute( "value", "3" );
+	const cat3Name = document.createTextNode( "cat3" );
+
+	const parentCat3 = document.createElement( "label" );
+	parentCat3.appendChild( cat3 );
+	parentCat3.appendChild( cat3Name );
+
+	const cat4 = document.createElement( "input" );
+	cat4.setAttribute( "type", "checkbox" );
+	cat4.setAttribute( "value", "4" );
+	const cat4Name = document.createTextNode( "Birds" );
+
+	const parentCat4 = document.createElement( "label" );
+	parentCat4.appendChild( cat4 );
+	parentCat4.appendChild( cat4Name );
+
+	const cat5 = document.createElement( "input" );
+	cat5.setAttribute( "type", "checkbox" );
+	cat5.setAttribute( "value", "5" );
+	const cat5Name = document.createTextNode( "dogs" );
+
+	const parentCat5 = document.createElement( "label" );
+	parentCat5.appendChild( cat5 );
+	parentCat5.appendChild( cat5Name );
+
+	const allCats = document.createElement( "div" );
+	allCats.setAttribute( "id", "categorychecklist" );
+	allCats.appendChild( parentCat1 );
+	allCats.appendChild( parentCat2 );
+	allCats.appendChild( parentCat3 );
+	allCats.appendChild( parentCat4 );
+	allCats.appendChild( parentCat5 );
+
+	const mostUsedCats = document.createElement( "div" );
+	mostUsedCats.setAttribute( "id", "categorychecklist-pop" );
+	mostUsedCats.appendChild( cat1.cloneNode() );
+	mostUsedCats.appendChild( cat2.cloneNode() );
```

Comment:
I'm not sure it this easily done, but I would split this up into two functions: one for creating the test data for all the categories and one for the most used categories.